segfast/segyio_loader.py

```diff
@@ -125,32 +122,40 @@ def load_headers(self, headers, reconstruct_tsf=True, sort_columns=True, tracewi
             If str, then type of progress bar to display: `'t'` for textual, `'n'` for widget.
         """
         _ = kwargs
-        headers = list(headers)
+        headers = self.make_headers(headers)
 
         if reconstruct_tsf and 'TRACE_SEQUENCE_FILE' in headers:
             headers.remove('TRACE_SEQUENCE_FILE')
 
-        headers_bytes = self.headers_to_bytes(headers)
-
         # Load data to buffer
         buffer = np.empty((self.n_traces, len(headers)), dtype=np.int32)
         if tracewise:
             for i, header_ in Notifier(pbar, total=self.n_traces, frequency=1000)(enumerate(self.file_handler.header)):
-                for j, byte_ in enumerate(headers_bytes):
-                    buffer[i, j] = header_.getfield(header_.buf, byte_)
+                for j, header in enumerate(headers):
+                    buffer[i, j] = header_.getfield(header_.buf, header.byte)
         else:
             for i, header in enumerate(headers):
                 buffer[:, i] = self.load_header(header)
 
         # Convert to pd.DataFrame, optionally add TSF and sort
-        dataframe = pd.DataFrame(buffer, columns=headers, copy=False)
+        dataframe = pd.DataFrame(buffer, columns=[item.name for item in headers], copy=False)
         dataframe = self.postprocess_headers_dataframe(dataframe, headers=headers,
                                                        reconstruct_tsf=reconstruct_tsf, sort_columns=sort_columns)
         return dataframe
 
+    def make_headers(self, headers):
```

Comment:
Looks like a proper place to pass loader endianness to initialize instances of `TraceHeades`